src/keychain_mac.cpp

```diff
@@ -111,28 +108,25 @@ void updateError(keychain::Error &err, OSStatus status) {
  */
 OSStatus modifyPassword(const std::string &serviceName, const std::string &user,
                         const std::string &password) {
-    SecKeychainItemRef item = NULL;
-    OSStatus status = SecKeychainFindGenericPassword(
-        defaultUserKeychain,
-        static_cast<UInt32>(serviceName.length()),
-        serviceName.data(),
-        static_cast<UInt32>(user.length()),
-        user.data(),
-        NULL, // unused output parameter
-        NULL, // unused output parameter
-        &item);
+    CFStringRef cfServiceName = CFStringCreateWithCString(NULL, serviceName.c_str(), kCFStringEncodingUTF8);
+    CFStringRef cfUser = CFStringCreateWithCString(NULL, user.c_str(), kCFStringEncodingUTF8);
```

Comment:
nitpick: let's use `kCFAllocatorDefault` instead of `NULL` here and in other `CF*Create*` calls, it's more explicit.